sbioapputils/app_runner/yaml_automation.py

```diff
@@ -170,20 +170,39 @@ def _prune_script(script_text):
     return new_text
 
 
-def _parse_input_python_v2(file: BytesIO, verbose: bool = False):
+def _prune_jupyter(script_text):
+    startstrings = ["#","'''","import","from"]
+    endstrings = ["'''"]
+    substrings = ["print","ipython"]
+    keep = ["="] #,"def ","class "]
+    new_text = ""
+    lines = list(set(script_text.splitlines()))
+    for line in lines:
+        if not any(line.lstrip().startswith(substring) for substring in startstrings):
+            if not any(line.rstrip().endswith(substring) for substring in endstrings):
+                if not any(substring in line for substring in substrings):
+                    if any(substring in line for substring in keep):
+                        new_text += line + "\n"
+    return(new_text)
+
+
+def _parse_input_python_v2(file: BytesIO, file_type: str = 'py', verbose: bool = False):
     script_text = file.getvalue().decode('ASCII')
-    stripped_script = _prune_script(script_text)
+    if file_type=='py':
+        stripped_script = _prune_script(script_text)
+    elif file_type=='ipynb':
+        stripped_script = _prune_jupyter(script_text)
     if verbose:
         line_count1 = len(script_text.splitlines())
         line_count2 = len(stripped_script.splitlines())
         print(f"Length of full script is {line_count1}, and the stripped script is {line_count2}")
     return stripped_script
 
 
-def _parse_multiple_files(file_list, verbose=False):
+def _parse_multiple_files(file_list, file_type: str = 'py', verbose=False):
```

Comment:
This function is called parse_multiple_files but it accepts the file_type which is confusing since different files might be different types